src/Data/DateTime.purs

```diff
@@ -0,0 +1,216 @@
+module Ocelot.Data.DateTime where
+
+import Prelude
+
+import Data.Array (drop, find, reverse, sort, take)
+import Data.Date (Date, Day, Year, canonicalDate, lastDayOfMonth, year, month, day, Weekday(..), weekday)
+import Data.DateTime (DateTime(..), Hour, Millisecond, Minute, Month(..), Second, Time(..))
+import Data.DateTime (adjust, date) as DT
+import Data.DateTime.Instant (fromDate, toDateTime)
+import Data.Either (either)
+import Data.Enum (class Enum, fromEnum, pred, succ, toEnum)
+import Data.Formatter.DateTime (Formatter)
+import Data.Formatter.DateTime as Formatter
+import Data.FunctorWithIndex (mapWithIndex)
+import Data.List (List(..), (:))
+import Data.Maybe (Maybe(..), fromJust, maybe)
+import Data.String.Regex (parseFlags, regex, replace)
+import Data.Time (Hour, Millisecond, Minute, Second, Time(..))
+import Data.Time (adjust) as T
+import Data.Time.Duration (Days(..), Hours(..), Minutes(..), negateDuration)
+import Data.Tuple (Tuple(..), fst, snd)
+import Partial.Unsafe (unsafePartial)
+
+
+----------
+-- Defaults
+defaultTime :: Time
+defaultTime = unsafeMkTime 0 0 0 0
+
+defaultDate :: Date
+defaultDate = unsafeMkDate 2000 1 1
+
+-- Array of 23 hour intervals, and one 59 min one
+defaultTimeRange :: Array Time
+defaultTimeRange = hourRange defaultTime 23 <> [prevMinute defaultTime]
+
+----------
+-- Time Ranges
+
+-- Generates an array of hours
+hourRange :: Time -> Int -> Array Time
+hourRange start hrs = go start 0 []
+  where
+    go cur count acc
+      | count == hrs = acc <> [cur]
+      | otherwise    = go (nextHour cur) (count + 1) (acc <> [cur])
+
+
+----------
+-- Date Ranges
+
+-- List each day in the month. Relies on `dateRange` under the hood.
+monthRange :: Year -> Month -> Array Date
+monthRange y m = dateRange start end
+  where
+    start = firstDateOfMonth y m
+    end   = lastDateOfMonth y m
+
+-- Provided two dates, generates the range (inclusive) between them.
+-- Note: Preserves ordering. If you provide a future then past date,
+-- your return range will be from the future to the past.
+dateRange :: Date -> Date -> Array Date
+dateRange = \s e -> go s e []
+  where
+    go start end acc
+      | start == end = reverse $ [start] <> acc
+      | start >  end = reverse $ go end start acc
+      | otherwise    = go (nextDay start) end ([start] <> acc)
+
+
+----------
+-- Time Manipulation
+nextHour :: Time -> Time
```

Comment:
no `prevHour` needed?

Reply:
Yeah I could add it for symmetry but didn't need it anywhere

Reply:
👍